v0/destinations/google_cloud_function/util.js

```diff
@@ -0,0 +1,97 @@
+const { defaultBatchRequestConfig, CustomError } = require("../../util");
+
+const { TRIGGERTYPE } = require("./config");
+
+const validateDestinationConfig = ({ Config }) => {
+  if (!Config.googleCloudFunctionUrl) {
+    throw new CustomError("[GCF]:: Url not found. Aborting", 400);
+  }
+  if (Config.TriggerType === "https") {
+    // Auth Mandatory
+    if (!Config.gcloudAuthorization) {
+      throw new CustomError("[GCF]:: Access Token not found. Aborting", 400);
+    }
+  }
+};
+
+function addHeader(response, Config) {
+  const { TriggerType, apiKeyId, gcloudAuthorization } = Config;
+  let basicAuth;
+  if (apiKeyId) {
+    basicAuth = Buffer.from(`apiKey:${apiKeyId}`).toString("base64");
+  }
+
+  response.headers = { "content-type": "application/json" };
+  if (TRIGGERTYPE.HTTPS === TriggerType.toLowerCase()) {
+    response.headers.Authorization = `bearer ${gcloudAuthorization}`;
+  }
+  if (basicAuth) {
+    if (TRIGGERTYPE.HTTPS === TriggerType.toLowerCase()) {
+      response.headers.ApiKey = `Basic ${basicAuth}`;
+    } else {
+      response.headers.Authorization = `Basic ${basicAuth}`;
+    }
+  }
+}
+
+function generateBatchedPayloadForArray(events) {
+  const batchResponseList = [];
+  const metadata = [];
+
+  // extracting destination
+  // from the first event in a batch
+  const { destination } = events[0];
+  const { googleCloudFunctionUrl } = destination.Config;
+  let batchEventResponse = defaultBatchRequestConfig();
+
+  // Batch event into dest batch structure
+  events.forEach(ev => {
+    batchResponseList.push(ev.message.body.JSON);
+    metadata.push(ev.metadata);
+  });
+
+  batchEventResponse.batchedRequest.body.JSON_ARRAY = {
+    batch: JSON.stringify(batchResponseList)
+  };
+
+  batchEventResponse.batchedRequest.endpoint = googleCloudFunctionUrl;
+
+  addHeader(batchEventResponse.batchedRequest, destination.Config);
+
+  batchEventResponse = {
+    ...batchEventResponse,
+    metadata,
+    destination
+  };
+
+  return batchEventResponse;
+}
+
+function generateBatchedPayload(event) {
+  // extracting destination
+  const { destination } = event;
+  const { googleCloudFunctionUrl } = destination.Config;
+  const metadata = [];
+  metadata.push(event.metadata);
+  let batchEventResponse = defaultBatchRequestConfig();
+
+  batchEventResponse.batchedRequest.body.JSON = event.message.body.JSON;
+
+  batchEventResponse.batchedRequest.endpoint = googleCloudFunctionUrl;
+
+  addHeader(batchEventResponse.batchedRequest, destination.Config);
```

Comment:
we are adding it thrice actually. Please edit it.

Reply:
resolved